hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/GenericsUtil.java

```diff
@@ -103,14 +113,15 @@ public static boolean isLog4jLogger(Class<?> clazz) {
    * @return true if the logger uses Log4J implementation.
    */
   public static boolean isLog4jLogger(String logger) {
-    if (logger == null) {
+    if (logger == null || !IS_LOG4J_LOGGER.get()) {
       return false;
     }
     Logger log = LoggerFactory.getLogger(logger);
     try {
```

Comment:
this try/catch is a duplicate of isLog4jLogger(class). 
why not do `isLog4jLogger(LoggerFactory.getLogger(logger))`

Reply:
I think the other way around would be more appropriate?
For instance, in slf4j implementation, they have:
```
    public static Logger getLogger(Class<?> clazz) {
        Logger logger = getLogger(clazz.getName());
...
...
```

Hence, we can also call clazz.getName() here.

Reply:
How about this latest commit? https://github.com/apache/hadoop/pull/5456/commits/c74b7a059a528369ba49ad279880a523d84d75cb

Reply:
Refactored to keep the logic in only one utility method. The other method will just call this one. Simplified (exactly as slf4j library methods).